Source/Core/DolphinTool/ConvertCommand.cpp

```diff
@@ -0,0 +1,276 @@
+      return 1;
+    }
+
+    if (!CheckBlockSize(block_size_o.value(), volume->GetSize(), format))
+    {
+      std::cerr << "Error: Block size is invalid" << std::endl;
+      return 1;
+    }
+  }
+
+  // --compress, --compress_level
+  std::optional<DiscIO::WIARVZCompressionType> compression_o =
+      ParseCompressionTypeString(static_cast<const char*>(options.get("compress")));
+
+  std::optional<int> compression_level_o;
+  if (options.is_set("compress_level"))
+    compression_level_o = static_cast<int>(options.get("compress_level"));
+
+  if (format == DiscIO::BlobType::WIA || format == DiscIO::BlobType::RVZ)
+  {
+    if (!compression_o.has_value() || !compression_level_o.has_value())
+    {
+      std::cerr << "Error: Compression format and level must be set for WIA or RVZ" << std::endl;
+      return 1;
+    }
+
+    const std::pair<int, int> range = DiscIO::GetAllowedCompressionLevels(compression_o.value());
+    if (compression_level_o.value() < range.first || compression_level_o.value() > range.second)
+    {
+      std::cerr << "Error: Compression not in acceptable range for the format" << std::endl;
+      return 1;
+    }
+  }
+
+  // Open the blob reader
+  std::unique_ptr<DiscIO::BlobReader> blob_reader =
+      scrub ? DiscIO::ScrubbedBlob::Create(input_file_path) :
+              DiscIO::CreateBlobReader(input_file_path);
+  if (!blob_reader)
+  {
+    std::cerr << "Error: Unable to process disc image. If --scrub is enabled, try again without it."
+              << std::endl;
+    return 1;
+  }
+
+  // Perform the conversion
+  const auto NOOP_STATUS_CALLBACK = [](const std::string& text, float percent) { return true; };
+
+  std::future<bool> success;
+
+  switch (format)
+  {
+  case DiscIO::BlobType::PLAIN:
+    success = std::async(std::launch::async, [&] {
+      const bool good = DiscIO::ConvertToPlain(blob_reader.get(), input_file_path, output_file_path,
+                                               NOOP_STATUS_CALLBACK);
+      return good;
+    });
+    break;
+
+  case DiscIO::BlobType::GCZ:
+    success = std::async(std::launch::async, [&] {
+      const bool good =
+          DiscIO::ConvertToGCZ(blob_reader.get(), input_file_path, output_file_path,
+                               volume->GetVolumeType() == DiscIO::Platform::WiiDisc ? 1 : 0,
+                               block_size_o.value(), NOOP_STATUS_CALLBACK);
+      return good;
+    });
+    break;
+
+  case DiscIO::BlobType::WIA:
+  case DiscIO::BlobType::RVZ:
+    success = std::async(std::launch::async, [&] {
+      const bool good = DiscIO::ConvertToWIAOrRVZ(
+          blob_reader.get(), input_file_path, output_file_path, format == DiscIO::BlobType::RVZ,
+          compression_o.value(), compression_level_o.value(), block_size_o.value(),
+          NOOP_STATUS_CALLBACK);
+      return good;
+    });
+    break;
+
+  default:
+    ASSERT(false);
+    break;
+  }
+
+  if (!success.get())
```

Comment:
If you're going to call `get` on the future immediately, there's no reason to use `std::launch::async`.

Reply:
My bad good call.